02-atmos/components/terraform/fetch-weather/main.tf

```diff
@@ -0,0 +1,47 @@
+variable "api_key" {
+  type        = string
+  description = "The API Key of for the openweathermap.org API"
+  sensitive   = true
+}
+
+variable "unit_of_measurement" {
+  default     = "imperial"
+  type        = string
+  description = <<-EOT
+  The unit of measurement to display the temperature in.
+  Valid values are `standard`, `metric`, or `imperial`
+  EOT
+}
+
+data "terraform_remote_state" "location" {
+  backend = "local"
+
+  config = {
+    path = "${path.root}/../fetch-location/terraform.tfstate.d/example/terraform.tfstate"
+  }
+}
+
+locals {
+  users_location_map = data.terraform_remote_state.location.outputs.users_location_map
+
+  city    = local.users_location_map.city
+  region  = local.users_location_map.region
+  country = local.users_location_map.country
+
+  url   = "https://api.openweathermap.org/data/2.5/weather"
+  query = "${local.city},${local.region},${local.country}"
```

Comment:
Maybe we can replace with `https://www.metaweather.com/api/location/search/?query=san` which doesn't require any keys?
https://www.metaweather.com/api/

Reply:
@osterman updated to this API and a more reliable IP Location API.